docs/agent/secrets.md

```diff
@@ -368,6 +421,85 @@ any secrets in it. This is done before setting up the logging. This means that
+setup. See Windows intructions [here](#windows).
+
+1. If any other group or user than needed has rights on the executable you will get a similar error in the log:
+   ```
+   error while decrypting secrets in an instance: Invalid executable 'C:\decrypt.exe': other users/groups than LOCAL_SYSTEM, Administrators or ddagentuser have rights on it
+   ```
+
+2. If `ddagentuser` doesn't have read and execute right on the file, you will get a similar error:
+   ```
+   error while decrypting secrets in an instance: could not query ACLs for C:\decrypt.exe
+   ```
+
+3. Your executable need to be a valid Win32 application, or you will get the following error:
+   ```
+   error while running 'C:\test.txt': fork/exec C:\decrypt.py: %1 is not a valid Win32 application.
+   ```
+
+###### Testing your executable
+
+Your executable will be executed by the agent when fetching your secrets. The
+Datadog agent run using the `ddagentuser`. This specific user has no specific
+rights but is part of the `Performance Monitor Users` group. The password for
+this user is randomly generated at install time and never saved anywhere.
+
+This means that your executable might work with your default user or
+development user but not when it's run by the agent, as `ddagentuser` has more
+restricted rights.
+
+The easiest way to test your executable in the same conditions as the agent is
+to update the password of the `ddagentuser` on your dev box. This way you will
+be able to authenticate as `ddagentuser` and run your executable in the same
+context the agent would.
+
+To do so, follow those steps:
+1. Remove `ddagentuser` from the `Local Policies/User Rights Assignement/Deny
+   Log on locally` list in the `Local Security Policy`.
+2. Set a new password for ddagentuser (since the one generated at install time
+   is never saved anywhere). In Powershell run:
+   ```powershell
+   $user = [ADSI]"WinNT://./ddagentuser";
+   $user.SetPassword("a_new_password")
+   ```
+3. Update the password to be used by `DatadogAgent` service in the Service
+   Control Manager. In powershell run:
+   ```powershell
+   sc.exe config DatadogAgent password= "a_new_password"
+   ```
+
+You can now login as `ddagentuser` to test your executable. We have a
+[powershell script](secrets_scripts/secrets_tester.ps1) to help you test your
+executable as another user. It will switch the user context and mimic how the
+agent runs your executable.
+
+Example on how to use it:
+```powershell
+.\secrets_tester.ps1 -user ddagentuser -password a_new_password -executable C:\path\to\your\executable.exe -payload '{"version": "1.0", "secrets": ["secret_ID_1", "secret_ID_2"]}'
+Creating new Process with C:\path\to\your\executable.exe
+Waiting a second for the process to be up and running
+Writing the payload to Stdin
+Waiting a second so the process can fetch the secrets
+stdout:
+{"secret_ID_1":{"value":"secret1"},"secret_ID_2":{"value":"secret2"}}
+stderr: None
+exit code:
+0
+```
+
+#### Agent refusing to start
```

Comment:
this section is duplicated (and the dupe above isn't in the right place)

Reply:
@hush-hush Can you have a look at this?  I suspect it was a botched merge...